framework/src/main/java/org/checkerframework/framework/type/HashcodeAtmVisitor.java

```diff
@@ -31,8 +31,10 @@ protected Integer defaultAction(AnnotatedTypeMirror type, Void v) {
         if (type == null) {
             return 0;
         }
+        if (type.underlyingTypeHashcode == -1) {
+            type.underlyingTypeHashcode = type.underlyingType.toString().hashCode() * 13;
```

Comment:
What is the point of this multiplication by 13?

Reply:
I think the caching code would be cleaner and easier to understand if it was performed in the class that has the cache.  That is, create a method on ATM like "getUnderlyingTypeHashCode".